crates/transaction-pool/src/test_utils/mock.rs

```diff
@@ -54,20 +54,20 @@ pub fn mock_tx_pool() -> MockTxPool {
 
 /// Sets the value for the field
 macro_rules! set_value {
-    ($this:ident => $field:ident) => {
+    ($this:expr => $field:ident) => {{
         let new_value = $field;
         match $this {
-            MockTransaction::Legacy { ref mut $field, .. } |
-            MockTransaction::Eip1559 { ref mut $field, .. } |
-            MockTransaction::Eip4844 { ref mut $field, .. } |
-            MockTransaction::Eip2930 { ref mut $field, .. } |
-            MockTransaction::Eip7702 { ref mut $field, .. } => {
-                *$field = new_value;
+            MockTransaction::Legacy { $field, .. } |
+            MockTransaction::Eip1559 { $field, .. } |
+            MockTransaction::Eip4844 { $field, .. } |
+            MockTransaction::Eip2930 { $field, .. } |
+            MockTransaction::Eip7702 { $field, .. } => {
+                *$field = new_value.into();
             }
         }
         // Ensure the tx cost is always correct after each mutation.
         $this.update_cost();
-    };
+    }};
```

Comment:
this doesn't compile

why do we need to change macro for 2024?

Reply:
from my chat with ai 

The issue is that the macro needs to handle different field types (u64, B256, Address, Bytes, etc.) and the match ergonomics are behaving differently in Rust 2024.

  1. In Rust 2021: The original code used ref mut $field in or-patterns, which explicitly created a mutable reference to the field regardless of its type.
  2. In Rust 2024:
    - ref mut in or-patterns is forbidden when matching on references
    - When matching &mut self, fields are automatically borrowed based on their usage
    - But the automatic borrowing behavior might not be consistent across all field types